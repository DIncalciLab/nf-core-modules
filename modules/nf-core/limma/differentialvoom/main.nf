process LIMMA_DIFFERENTIALVOOM {
    tag "$meta"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'oras://community.wave.seqera.io/library/bioconductor-edger_bioconductor-ihw_bioconductor-limma_r-dplyr_r-readr:7fc48564d286c1c6' :
        'community.wave.seqera.io/library/bioconductor-edger_bioconductor-ihw_bioconductor-limma_r-dplyr_r-readr:edea0f9fbaeba3a0' }"

    input:
    tuple val(meta), val(contrast_variable), val(reference), val(target)
    tuple val(meta2), path(samplesheet), path(intensities)

    output:
    tuple val(meta), path("*.limma.results.tsv")          , emit: results
    tuple val(meta), path("*.limma.mean_difference.png")  , emit: md_plot
    tuple val(meta), path("*.MArrayLM.limma.rds")         , emit: rdata
    tuple val(meta), path("*.normalised_counts.tsv")      , emit: normalised_counts, optional: true
    tuple val(meta), path("*.limma.model.txt")            , emit: model
    tuple val(meta), path("*.R_sessionInfo.log")          , emit: session_info
    path "versions.yml"                                   , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    template 'limma_de.R'

    stub:
    prefix              = task.ext.prefix   ?: "${meta.id}"
    """
    #!/usr/bin/env Rscript
    library(limma)

    a <- file("${prefix}.limma.results.tsv", "w")
    close(a)
    a <- file("${prefix}.limma.mean_difference.png", "w")
    close(a)
    a <- file("${prefix}.MArrayLM.limma.rds", "w")
    close(a)
    a <- file("${prefix}.normalised_counts.tsv", "w")
    close(a)
    a <- file("${prefix}.limma.model.txt", "w")
    close(a)
    a <- file("${prefix}.R_sessionInfo.log", "w")
    close(a)

    ## VERSIONS FILE
    r.version <- strsplit(version[['version.string']], ' ')[[1]][3]
    limma.version <- as.character(packageVersion('limma'))

    writeLines(
        c(
            '"task.process":',
            paste('    r-base:', r.version),
            paste('    bioconductor-limma:', limma.version)
        ),
        'versions.yml'
    )
    """
}
