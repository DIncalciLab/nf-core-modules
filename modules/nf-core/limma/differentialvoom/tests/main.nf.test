// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test limma/differentialvoom
nextflow_process {

    name "Test Process LIMMA_DIFFERENTIALVOOM"
    script "../main.nf"
    process "LIMMA_DIFFERENTIALVOOM"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "limma"
    tag "limma/differentialvoom"

    test("test_limma_differential") {

        when {
            process {
                """
                input[0] = Channel.of(['id': 'test', 'variable': 'treatment', 'reference': 'hND6', 'target': 'mCherry'])
                        .map{tuple(it, it.variable, it.reference, it.target)}
                input[1] = Channel.of([[id:'test'],
                                      file(params.modules_testdata_base_path + 'genomics/mus_musculus/rnaseq_expression/SRP254919.samplesheet.csv', checkIfExists: true),
                                      file(params.modules_testdata_base_path + 'genomics/mus_musculus/rnaseq_expression/SRP254919.salmon.merged.gene_counts.top1000cov.tsv', checkIfExists: true)])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_limma_differential - stub") {

        options "-stub"

        when {
            process {                """
                input[0] = Channel.of(['id': 'test', 'variable': 'treatment', 'reference': 'hND6', 'target': 'mCherry'])
                        .map{tuple(it, it.variable, it.reference, it.target)}
                input[1] = Channel.of([[id:'test'],
                                      file(params.modules_testdata_base_path + 'genomics/mus_musculus/rnaseq_expression/SRP254919.samplesheet.csv', checkIfExists: true),
                                      file(params.modules_testdata_base_path + 'genomics/mus_musculus/rnaseq_expression/SRP254919.salmon.merged.gene_counts.top1000cov.tsv', checkIfExists: true)])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}