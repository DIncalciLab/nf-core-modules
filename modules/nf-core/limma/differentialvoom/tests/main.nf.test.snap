{
    "test_limma_differential - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.results.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.mean_difference.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.MArrayLM.limma.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.normalised_counts.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.model.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.R_sessionInfo.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "6": [
                    "versions.yml:md5,c90309ffa2653f7204690ea14064d2e9"
                ],
                "md_plot": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.mean_difference.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "model": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.model.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "normalised_counts": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.normalised_counts.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "rdata": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.MArrayLM.limma.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "results": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.results.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "session_info": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.R_sessionInfo.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c90309ffa2653f7204690ea14064d2e9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-14T23:12:18.543218"
    },
    "test_limma_differential": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.results.tsv:md5,5ac6b133d00f25a8623dec9f38af98e4"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.mean_difference.png:md5,5dd9b0185b917dd6f9e0506d71705421"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.MArrayLM.limma.rds:md5,23543e8796931ec2fce8600c4f5a4606"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.normalised_counts.tsv:md5,a9c81ca72e246e78505e4bf4ca1d1df4"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.model.txt:md5,37fb2d77cd222b9cddaa38e8e6dddaf8"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.R_sessionInfo.log:md5,e51f5a47c37d58cf4e8a0c6766759736"
                    ]
                ],
                "6": [
                    "versions.yml:md5,c90309ffa2653f7204690ea14064d2e9"
                ],
                "md_plot": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.mean_difference.png:md5,5dd9b0185b917dd6f9e0506d71705421"
                    ]
                ],
                "model": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.model.txt:md5,37fb2d77cd222b9cddaa38e8e6dddaf8"
                    ]
                ],
                "normalised_counts": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.normalised_counts.tsv:md5,a9c81ca72e246e78505e4bf4ca1d1df4"
                    ]
                ],
                "rdata": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.MArrayLM.limma.rds:md5,23543e8796931ec2fce8600c4f5a4606"
                    ]
                ],
                "results": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.limma.results.tsv:md5,5ac6b133d00f25a8623dec9f38af98e4"
                    ]
                ],
                "session_info": [
                    [
                        {
                            "id": "test",
                            "variable": "treatment",
                            "reference": "hND6",
                            "target": "mCherry"
                        },
                        "test.R_sessionInfo.log:md5,e51f5a47c37d58cf4e8a0c6766759736"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c90309ffa2653f7204690ea14064d2e9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-14T23:12:08.333183"
    }
}